tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py

```diff
@@ -0,0 +1,195 @@
+            cephfs_rwx_pvcs, self.namespace, constants.CEPHFS_INTERFACE)
+        rbd_rwo_pods = helpers.create_pods_parallel(
+            rbd_rwo_pvcs, self.namespace, constants.CEPHBLOCKPOOL)
+        # TODO: RBD RWX pod creaiton
+        # rbd_rwx_pods = helpers.create_pods_parallel(
+        #     rbd_rwx_pvcs, self.namespace, constants.CEPHBLOCKPOOL)
+        temp_pod_objs.extend(cephfs_rwo_pods + cephfs_rwx_pods + rbd_rwo_pods)
+        # Appending all the pvc obj to base case param for cleanup and evaluation
+        self.all_pvc_obj.extend(cephfs_rwo_pvcs + cephfs_rwx_pvcs + rbd_rwo_pvcs + rbd_rwx_pvcs)
+        self.all_pod_obj.extend(temp_pod_objs)
+        # IO will start based on TC requirement
+        if self.run_io == 'Yes':
+            with ThreadPoolExecutor() as executor:
+                for pod_obj in temp_pod_objs:
+                    executor.submit(pod_obj.run_io('fs', size='1G'))
+
+        log.info(f"all_pod_list {self.all_pod_obj}")
+        log.info(f"all_pvc_list {self.all_pvc_obj}")
+        log.info(f"delete_pod_count {self.delete_pod_count}")
+
+    def delete_pvc_pod(self):
+        """
+        Function to delete pvc and pod based on the delete pod count.
+        """
+        log.info(f"Delete {self.delete_pod_count} pods and respective pvcs")
+        temp_pod_list = random.choices(self.all_pod_obj, k=self.delete_pod_count)
+        temp_pvc_list = []
+        for pod_obj in temp_pod_list:
+            for pvc_obj in self.all_pvc_obj:
+                if pod_obj.pvc_name == pvc_obj.name:
+                    temp_pvc_list.append(pvc_obj)
+                    log.info(f"Deleting pvc {pvc_obj.name}")
+                    self.all_pvc_obj.remove(pvc_obj)
+            log.info(f"Deleting pod {pod_obj}")
+            if pod_obj in self.all_pod_obj:
+                self.all_pod_obj.remove(pod_obj)
+        helpers.delete_objs_parallel(temp_pod_list)
+        helpers.delete_objs_parallel(temp_pvc_list)
+
+    def cleanup(self):
+        """
+        Function to cleanup the SC, PVC and POD objects parallel.
+        """
+        helpers.delete_objs_parallel(self.all_pod_obj)
+        helpers.delete_objs_parallel(self.all_pvc_obj)
+        self.rbd_sc_obj.delete()
+        self.cephfs_sc_obj.delete()
+
+
+@scale
+@pytest.mark.usefixtures(
+    create_project.__name__,
+    create_rbd_secret.__name__,
+    create_cephfs_secret.__name__,
+    create_ceph_block_pool.__name__
+)
+class TestPVSTOcsCreateDeletePVCsWithAndWithoutIO(BasePvcPodCreateDelete):
+    """
+    Class for TC OCS-682 & OCS-679 Create & Delete Cluster with 1000 PVC with
+    and without IO, then Increase the PVC count to 1500. Check for Memory leak
+    """
+    @pytest.fixture()
+    def setup_fixture(self, request):
+        def finalizer():
+            self.cleanup()
+
+        request.addfinalizer(finalizer)
+
+    @pytest.mark.parametrize(
+        argnames="start_io",
+        argvalues=[
+            pytest.param(
+                *['No'],
```

Comment:
You can use True/False here instead of 'Yes' and 'No'

Reply:
done,